### DISCUSSION TITLE
Windows Support

### DISCUSSION THREAD

**Discussion by achanin00** [SthPhoenix/InsightFace-REST#110] [Q&A]
Can this project be built under Windows?  

deploy_cpu.sh and deploy_trt.sh appear to be Linux only.

**Comment by SthPhoenix** [comment]
Hi! This project wasn't tested on Windows, but i think you can run CPU version without docker if you install all requirements from `src/requirements.txt`

CPU version uses `onnxruntime` for inference, but there is also a `onnxruntime-gpu` which should work on Windows, but you should also modify [src/api_trt/modules/model_zoo/exec_backends/onnxrt_backend.py](https://github.com/SthPhoenix/InsightFace-REST/blob/master/src/api_trt/modules/model_zoo/exec_backends/onnxrt_backend.py) by replacing lines 12, 39, 80 with:
`self.rec_model = onnxruntime.InferenceSession(rec_name, providers=['CUDAExecutionProvider', 'CPUExecutionProvider'])`

and line 117 with:
`self.rec_model = onnxruntime.InferenceSession(model, providers=['CUDAExecutionProvider', 'CPUExecutionProvider'])`

Then use [src/api_trt/entrypoint.sh](https://github.com/SthPhoenix/InsightFace-REST/blob/master/src/api_trt/entrypoint.sh) as reference for commands required to start the service.

TRT version won't work on Windows because AFAIK there is no TRT Python wrapper for windows. 

Also there were reports that at least CPU version works on Windows using Docker in WLS2, GPU version [wasn't working](https://github.com/SthPhoenix/InsightFace-REST/issues/58) last year, but maybe NVIDIA or Microsoft has fixed compatibility issues over the year

**Comment by achanin00** [comment]
Thank you for the pointers.  I will give it a try.
